### PR TITLE
Hover effect for regions

### PR DESCRIPTION
**Pull request by AntonRaichuk**

### CODE REVIEW

File: src/others/components/map/Map.tsx [hunk 1]
```diff
@@ -30,12 +30,21 @@ export const Map = ({ sourceWithLayer, interactiveLayerIds }: MapProps) => {
...
   const mapRef = useRef<MapRef>(null);
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
+  const [hoveredRegionId, setHoveredRegionId] = useState<number | string | undefined>(undefined);
   const [cursor, setCursor] = useState<"auto" | "pointer">("auto");
 
   const closePopup = useCallback(() => {
     setCursor("auto");
     setPopupInfo(null);
-  }, []);
+    if (mapRef?.current) {
```
Comment (alisonelizabeth) [inline src/others/components/map/Map.tsx:39]: [nit] In https://github.com/ukraine-taskforce/requests-dashboard/pull/71, I wrapped the page in `MapProvider` so we have access to the [`useMap()`](https://visgl.github.io/react-map-gl/docs/api-reference/use-map) hook, so we don't need to reference the `ref` anymore.

File: src/others/components/map/Map.tsx [hunk 2]
```diff
@@ -30,12 +30,21 @@ export const Map = ({ sourceWithLayer, interactiveLayerIds }: MapProps) => {
   const { t } = useTranslation();
   const mapRef = useRef<MapRef>(null);
   const [popupInfo, setPopupInfo] = useState<PopupInfo | null>(null);
+  const [hoveredRegionId, setHoveredRegionId] = useState<number | string | undefined>(undefined);
```
Comment (alisonelizabeth) [inline src/others/components/map/Map.tsx:33]: In what situation will the region id be a number vs. a string?

**Review by alisonelizabeth — APPROVED** [review]
Nice work @AntonRaichuk! The hover works as expected. Left a couple minor code comments.

In terms of if I find it useful - I find the popover a little bit jumpy for regions, and it's often large due to the length of items (unrelated to your change, of course). However, this makes it hard for me to gain much value from the hover effect as the region is sometimes partially blocked from the popover, and my initial focus was of the popover. Let me know what you think!

File: src/others/components/map/Map.tsx (hunk 1 above)
Comment (AntonRaichuk) [reply src/others/components/map/Map.tsx]: done

**Comment by AntonRaichuk** [comment]
> Nice work @AntonRaichuk! The hover works as expected. Left a couple minor code comments.
> 
> In terms of if I find it useful - I find the popover a little bit jumpy for regions, and it's often large due to the length of items (unrelated to your change, of course). However, this makes it hard for me to gain much value from the hover effect as the region is sometimes partially blocked from the popover, and my initial focus was of the popover. Let me know what you think!

Yeah, that's exactly the point, that with our popup this hover effect does not seem to matter much.
So I'm quite hesitant if we should actually merge it (additional complexity for no clear benefit).

**Review by jciepka — APPROVED** [review]

**Pull request closed without merging**